### PR TITLE
Adding rule checking ViewEncapsulation

### PR DESCRIPTION
**Pull request by GregOnNet**
Hi,

this PR adds the rule `use-view-encapsulation` as requested in #300.
Please let me know if I can improve the code of this rule.

image
*rule tested in VS Code*

Kind Regards
Gregor

### CODE REVIEW

File: src/useViewEncapsulationRule.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    ruleName: 'use-view-encapsulation',
+    type: 'maintainability',
+    description: 'Disallows using of ViewEncapsulation.None',
+    rationale: '',
+    options: null,
+    optionsDescription: 'Not configurable',
+    typescriptOnly: true
+  };
+
+  static FAILURE = 'Using "ViewEncapsulation.None" may cause conflicts between css rules having the same selector';
```
Comment (mgechev) [inline src/useViewEncapsulationRule.ts]: Let's change the warning to: `'Using "ViewEncapsulation.None" will make your styles global which may have unintended effect'`.

**Comment by mgechev** [comment]
@GregOnNet good job! I added a minor comment regarding the warning message.

**Comment by GregOnNet** [comment]
@mgechev Thanks for your feedback.
I have updated the warning message as requested.
If there is anything else please let me know. :-)

File: src/useViewEncapsulationRule.ts (hunk 1 above)
Comment (GregOnNet) [inline src/useViewEncapsulationRule.ts]: Thank you. Please refer to 88e3cd417204b063e1bd2155c29de3bd7a2132fa to see the updated warning.

**Pull request merged**